docs/admin/node.md

```diff
@@ -120,25 +120,52 @@ Currently, there are three components that interact with the Kubernetes node int
+to ConditionUnknown when a node becomes unreachable (i.e. node controller stops receiving heartbeats
+e.g. due to the node being down), and then later evicting all the pods from the node
+(using graceful termination) if the node continues to be unreachable (the current timeouts
+are 40s to start reporting ConditionUnknown and 5m after that to start evicting pods).
+Node controller checks the state of each node every `--node-monitor-period` seconds.
+
+In 1.4 release we updated the logic of node controller to better handle cases when a
+big number of Nodes have problems with reaching the master machine (e.g. because
+master machine has networking problem). Starting with 1.4 node controller will look at the
+state of all nodes in the cluster when making a decision about pod eviction. In previous
+releases if the master machine got partitioned from the cluster for at least 5m40s,
+it would delete all running pods from its "desired state" for the cluster,
+and then when the partition ended and the kubelets were able to contact the master again,
+they would see all of their pods had been deleted and would kill them. Since 1.4 this shouldn't happen.
+
+In normal circumstances (read: other than ones described in following paragraphs)
+node controller limits the eviction rate to `--node-eviction-rate` (default 0.1)
+per second, i.e. it won't evict pods from more than 1 node per 10 seconds.
+
+When a node in a given availability zone (note: if your cluster does not span multiple cloud
+provider availability zones, then there is only one availability zone, namely the whole cluster)
+becomes unhealthy, node controller checks what percentage of nodes in the zone are unhealthy
+(NodeReady condition becomes ConditionUnknown or ConditionFalse) at the same time. If the
+fraction of unhealthy nodes is at least `--unhealthy-zone-threshold` (default 0.55) then
+the eviction rate is  reduced: if the cluster is small (i.e. has less than or equal to
+`--large-cluster-size-threshold` nodes - default 50) then evictions are stopped, otherwise
+the eviction rate is reduced to `--secondary-node-eviction-rate` (default 0.01) per second.
+The reason these policies are implemented per availability zone is because one availability
+zone might become partitioned from the master while the others remain connected.
+
+A key reason for spreading your nodes across availability zones is so that workload can be
+shifted to healthy zones when one entire zone goes down. To enable this behavior, if all
+nodes in a zone are unhealthy then node controller evicts at the normal rate `--node-eviction-rate`.
+The corner case for that is when all zones are completely unhealthy (i.e. there's no healthy node in
+the cluster). In such case node controller assumes that there's some problem with master machine
+connectivity and stops all evictions until any connectivity is restored.
+
+Third responsibility of node controler is keeping the list of nodes up to date with
```

Comment:
I would rephrase this as "Third responsibility of node controller is keeping the node controller's list of nodes up to date with the cloud provider's list of available machines."

And actually I would make this second, and move the eviction stuff to third (so it's the last part of this section).